frontend/apps/web/app/(mgmt)/[account]/new/job/connect/page.tsx

```diff
@@ -576,3 +718,76 @@ export default function Page({ searchParams }: PageProps): ReactElement {
     </div>
   );
 }
+
+interface TestConnectionBadgeProps {
+  isValidating: boolean;
+  validationResponse: CheckConnectionConfigResponse | undefined;
+  id: string | undefined;
+  accountName: string;
+}
+
+function TestConnectionBadge(props: TestConnectionBadgeProps): ReactElement {
+  const { isValidating, validationResponse, id, accountName } = props;
+
+  const url = `/${accountName}/connections/${id}/permissions`;
+
+  const handleValidationResponse = (
```

Comment:
Probably worth moving this outside of this function to be its own standalone component that is isolated, that way it doesnt have to be rebuilt every time and react can better render it.